master/pkg/searcher/grid.go

```diff
@@ -61,13 +61,31 @@ func (s *gridSearch) initialOperations(ctx context) ([]Operation, error) {
 	return ops, nil
 }
 
-func (s *gridSearch) progress(trialProgress map[model.RequestID]model.PartialUnits) float64 {
+func (s *gridSearch) progress(
+	trialProgress map[model.RequestID]model.PartialUnits,
+	trialsClosed map[model.RequestID]bool) float64 {
 	if s.MaxConcurrentTrials() > 0 && s.PendingTrials > s.MaxConcurrentTrials() {
 		panic("pending trials is greater than max_concurrent_trials")
 	}
-	unitsCompleted := sumTrialLengths(trialProgress)
+	unitsCompleted := 0.
+	for k, v := range trialsClosed {
+		if _, ok := trialProgress[k]; !ok {
+			if v {
+				unitsCompleted += float64(s.MaxLength().Units)
+			}
+		}
+	}
+	for k, v := range trialProgress {
+		if closed, ok := trialsClosed[k]; ok {
+			if closed {
+				unitsCompleted += float64(s.MaxLength().Units)
+			}
+		} else {
+			unitsCompleted += float64(v)
+		}
+	}
```

Comment:
i think this an equivalent and much easier to read version of this logic. trialsClosed.where(has no progress) union trialProgress.where(closed) is the same set as just trialsClosed.
```suggestion
	// Count each closed trial completely, no matter how it closed (invalid HP counts).
	for range trialsClosed {
		unitsCompleted += float64(s.MaxLength().Units)
	}
	// Count not closed trials for what they have completed.
	for _, v := range trialProgress {
		unitsCompleted += float64(v)
	}
```

Reply:
it might just be the set map value checks. it's 2 extra things to think about.
```suggestion
for k := range trialsClosed {
		if _, ok := trialProgress[k]; !ok {
			unitsCompleted += float64(s.MaxLength().Units)
		}
	}
	for k, v := range trialProgress {
		if trialsClosed[k] {
			unitsCompleted += float64(s.MaxLength().Units)
		} else {
			unitsCompleted += float64(v)
		}
	}
```